sql/hive/src/main/scala/org/apache/spark/sql/hive/SQLBuilder.scala

```diff
@@ -129,10 +129,13 @@ class SQLBuilder(logicalPlan: LogicalPlan, sqlContext: SQLContext) extends Loggi
         conditionSQL = condition.sql
       } yield s"$childSQL $whereOrHaving $conditionSQL"
 
-    case Union(left, right) =>
+    case Union(children) if children.length == 1 =>
+      toSQL(children.head)
```

Comment:
We still need it for handling the last child of union since another case for union is dependent on it. Not sure if this is the best way to handle it. I added a comment in the code to explain this issue. Thank you!